### ISSUE TITLE
Removing rpm-devel dependency

### ISSUE THREAD

**Issue by junaruga** [junaruga/rpm-py-installer#70]
From https://pagure.io/koji/issue/458#comment-467020

Right now `rpm-devel` is required to build the RPM Python binding.
to use `/usr/lib64/pkgconfig/rpm.pc` as a RPM's default behavior.

But maybe it is possible not to use this file.

> if not, just print some warning and let installation continue?

Implement this.

**Comment by junaruga** [comment]
This was kind of bug.
`setup.py` checking parent directory's `rpm.pc` from `pkgconfig` command in `setup.py`.
https://github.com/rpm-software-management/rpm/blob/rpm-4.14.x/python/setup.py.in#L25

But in `rpm-py-installer`'s install process, there is no `rpm.pc` in the parent directory.

So, the solution is creating from `rpm.pc.in` to `rpm.pc` manually in the install process.

**Comment by junaruga** [comment]
Change env name from `SETUP_PY_OPTM` to `RPM_PY_OPTM`.

And the logic is something like

```
if RPM_PY_OPTM == true:
  try with optimized mode.
    failed => try without optimized mode.
if RPM_PY_OPTM == false:
   try without optimized mode
```

**Comment by junaruga** [comment]
It is possible not to use `rpm-devel` for the install process.

Several ways to achieve it.
First of all, it is possible to install without `rpm-devel` after researching.
`rpm-libs` and `rpm-build-libs` are still needed.
But those are installed as a default, when I checked CentOS and Fedora container.

However using `rpm.pc` is more reliable than creating `rpm.pc` manually to predicting target condition.

So, I would try below logic to install.

1. Check if `rpm-devel` is installed. `rpm -q rpm-devel`.
1-1. If `rpm-devel` is installed, use it.
1-2. If `rpm-devel` is not installed, move on step 2.

2. Check current system is working by `dnf` or `yum`.

3. Check if the download plugin (feature) is available.

```
$ dnf download rpm-devel
```

or 

```
$ yum install --downloadonly --downloaddir=. rpm-devel
```

3-1. If the download plugin is installed, download `rpm-devel`, and extract and use `rpm-devel`.
3-2. If the download plugin is not installed, move on step 4.

4. manually create `rpm.pc`, and build and install it.
4.1. If the install is succeeded, the install process is done.
4.2. If the install is failed, the `rpm-py-installer` requires `rpm-devel`. And try to install with the `rpm-devel`.

**Comment by junaruga** [comment]
This issue is going to be fixed and released on next minor version release `v0.5.0`.
I am working on https://github.com/junaruga/rpm-py-installer/tree/feature/remove-rpm-devel-dependency branch now.
